lib/FusionInventory/Agent/Tools/Hardware.pm

```diff
@@ -1569,6 +1639,75 @@ sub _getPAGPInfo {
     return $results;
 }
 
+sub _setPhysicalComponents {
+    my (%params) = @_;
+
+    my $device = $params{device};
+    my $snmp   = $params{snmp};
+    my $logger = $params{logger};
+
+    # $components is a sparse hash of physical components, indexed by identifier
+    # (entPhysicalIndex)
+    my $components;
+
+    foreach my $key (keys %physical_components_variables) {
+        my $variable = $physical_components_variables{$key};
+        next unless $variable->{oid};
+        
+        my $results;
+        if (ref $variable->{oid} eq 'ARRAY') {
+            foreach my $oid (@{$variable->{oid}}) {
+                $results = $snmp->walk($oid);
+                last if $results;
+            }
+        } else {
+            $results = $snmp->walk($variable->{oid});
+        }
+
+        if ($key eq 'TYPE') {
+            while (my ($suffix, $raw_value) = each %{$results}) {
+                my $value =
+                    $raw_value eq 'other(1)'       ? 'other' :
+                    $raw_value eq '1'              ? 'other' :
+                    $raw_value eq 'unknown(2)'     ? 'unknown' :
+                    $raw_value eq '2'              ? 'unknown' :
+                    $raw_value eq 'chassis(3)'     ? 'chassis' :
+                    $raw_value eq '3'              ? 'chassis' :
+                    $raw_value eq 'backplane(4)'   ? 'backplane' :
+                    $raw_value eq '4'              ? 'backplane' :
+                    $raw_value eq 'container(5)'   ? 'container' :
+                    $raw_value eq '5'              ? 'container' :
+                    $raw_value eq 'powerSupply(6)' ? 'powerSupply' :
+                    $raw_value eq '6'              ? 'powerSupply' :
+                    $raw_value eq 'fan(7)'         ? 'fan' :
+                    $raw_value eq '7'              ? 'fan' :
+                    $raw_value eq 'sensor(8)'      ? 'sensor' :
+                    $raw_value eq '8'              ? 'sensor' :
+                    $raw_value eq 'module(9)'      ? 'module' :
+                    $raw_value eq '9'              ? 'module' :
+                    $raw_value eq 'port(10)'       ? 'port' :
+                    $raw_value eq '10'             ? 'port' :
+                    $raw_value eq 'stack(11)'      ? 'stack' :
+                    $raw_value eq '11'             ? 'stack' :
+                                                $raw_value;
```

Comment:
such a long list of consecutive checks is expensive, especially inside a loop. An hash-based solution would be better, ie:

outside the loop:
```
my %types = (
    1           => 'other',
    other(1) => 'other',
   ...
);
```

inside the loop:
`my $value = $types{$raw_value} || $raw_value;`